crates/binjs_io/src/entropy/write/lazy_stream.rs

```diff
@@ -76,67 +95,93 @@ impl LazyStream {
 
     /// Return the number of bytes written so far.
     pub fn bytes_written(&self) -> usize {
-        self.bytes_written
+        self.buffer.len()
     }
 
     /// Get the file for writing, initializing it if necessary.
     ///
     /// Always returns Ok(None) if `dump_path` was specified as `None`.
-    fn get_file(&mut self) -> Result<Option<&mut std::fs::File>, std::io::Error> {
-        if let Some(ref mut writer) = self.lazy_dump_file {
-            return Ok(Some(writer));
+    fn get_file(&mut self) -> Result<Option<&mut Dumps>, std::io::Error> {
+        if let Some(ref mut dumps) = self.lazy_dumps {
+            return Ok(Some(dumps));
         }
-        if let Some(ref path) = self.dump_path {
-            let dir = path.parent().unwrap();
-            std::fs::DirBuilder::new().recursive(true).create(dir)?;
-            let file = std::fs::File::create(path)?;
-            self.lazy_dump_file = Some(file);
-            return Ok(Some(self.lazy_dump_file.as_mut().unwrap()));
+        if let Some(mut path) = self.dump_path.take() {
+            {
+                let dir = path.parent().unwrap();
+                std::fs::DirBuilder::new().recursive(true).create(dir)?;
+            }
+
+            let raw = std::fs::File::create(&path)?;
+
+            let extension = match path.extension() {
```

Comment:
Technically, `.extension()` / `.set_extension()` are meant to be used for a single extension, which is why this becomes a bit verbose. For our case, maybe it would be simpler to use string-based manipulation here?

```rust
let mut path = path.into_os_string();
path.push(".bro");
```

Reply:
I have no strong opinion on this – especially since this code is dedicated to debugging/forensics.

Reply:
I'm mostly suggesting this for readability (the string version seems simpler), but it's up to you.